diff --git a/node_modules/vite-plugin-html/dist/index.cjs b/node_modules/vite-plugin-html/dist/index.cjs
index 704d055..a5b77c9 100644
--- a/node_modules/vite-plugin-html/dist/index.cjs
+++ b/node_modules/vite-plugin-html/dist/index.cjs
@@ -118,7 +118,8 @@ function createPlugin(userOptions = {}) {
         _pages = pages.map((page) => {
           return {
             filename: page.filename || DEFAULT_TEMPLATE,
-            template: page.template || DEFAULT_TEMPLATE
+            template: page.template || DEFAULT_TEMPLATE,
+            depFiles: page.depFiles ?? []
           };
         });
       }
@@ -126,30 +127,55 @@ function createPlugin(userOptions = {}) {
       const baseUrl = viteConfig.base ?? "/";
       const keys = Object.keys(proxy);
       let indexPage = null;
+
+      /** @type {{
+        filename: string;
+        depFiles: (string | RegExp)[];
+      }[]} */
+      let reloadEffects = [];
       for (const page of _pages) {
+        reloadEffects.push({
+          filename: page.filename,
+          depFiles: page.depFiles
+        });
         if (page.filename !== "index.html") {
-          rewrites.push(createRewire(page.template, page, baseUrl, keys));
+          rewrites.push(createRewire(baseUrl + page.filename, page, baseUrl, keys));
         } else {
           indexPage = page;
         }
       }
       if (indexPage) {
-        rewrites.push(createRewire("", indexPage, baseUrl, keys));
+        rewrites.push(createRewire(`${baseUrl}(index(.html)?)?`, indexPage, baseUrl, keys));
       }
       server.middlewares.use(history__default({
         disableDotRule: void 0,
         htmlAcceptHeaders: ["text/html", "application/xhtml+xml"],
         rewrites
       }));
+      server.watcher.on('change', file => {
+        reloadEffects.forEach(({ filename, depFiles }) => {
+          if (depFiles.some(depFile => {
+            if (typeof depFile === 'string') {
+              return file.includes(depFile)
+            } else {
+              return depFile.test(file)
+            }
+          })) {
+            server.ws.send({
+              type: 'full-reload',
+              path: filename === 'index.html' ? '*' : `${baseUrl}${filename}*`
+            })
+          }
+        })
+      });
     },
     transformIndexHtml: {
       enforce: "pre",
       async transform(html, ctx) {
-        const url = ctx.filename;
+        const url = ctx.originalUrl;
         const base = viteConfig.base;
-        const excludeBaseUrl = url.replace(base, "/");
-        const htmlName = path__default.relative(process.cwd(), excludeBaseUrl);
-        const page = getPage(userOptions, htmlName, viteConfig);
+        const excludeBaseUrl = url.replace(base, "");
+        const page = getPage(userOptions, excludeBaseUrl, viteConfig);
         const { injectOptions = {} } = page;
         const _html = await renderHtml(html, {
           injectOptions,
@@ -227,7 +253,7 @@ async function renderHtml(html, config) {
     ...viteConfig?.env ?? {},
     ...viteConfig?.define ?? {},
     ...env || {},
-    ...data
+    ...(typeof data === 'function' ? await data() : data)
   };
   let result = await ejs.render(html, ejsData, ejsOptions);
   if (entry) {
@@ -279,13 +305,13 @@ function getPageConfig(htmlName, pages, defaultPage) {
     template: `./${defaultPage}`
   };
   const page = pages.filter((page2) => {
-    return path__default.resolve("/" + page2.template) === path__default.resolve("/" + htmlName);
+    return path__default.resolve("/" + page2.filename) === path__default.resolve("/" + htmlName);
   })?.[0];
   return page ?? defaultPageOption ?? void 0;
 }
 function createRewire(reg, page, baseUrl, proxyUrlKeys) {
   return {
-    from: new RegExp(`^/${reg}*`),
+    from: new RegExp(`${reg}.*`),
     to({ parsedUrl }) {
       const pathname = parsedUrl.pathname;
       const excludeBaseUrl = pathname.replace(baseUrl, "/");
diff --git a/node_modules/vite-plugin-html/dist/index.d.ts b/node_modules/vite-plugin-html/dist/index.d.ts
index 0925ff6..b6c8e82 100644
--- a/node_modules/vite-plugin-html/dist/index.d.ts
+++ b/node_modules/vite-plugin-html/dist/index.d.ts
@@ -6,7 +6,7 @@ interface InjectOptions {
     /**
      *  @description Data injected into the html template
      */
-    data?: Record<string, any>;
+    data?: Record<string, any> | (() => Record<string, any>);
     tags?: HtmlTagDescriptor[];
     /**
      * @description esj options configuration
@@ -18,6 +18,7 @@ interface PageOption {
     template: string;
     entry?: string;
     injectOptions?: InjectOptions;
+    depFiles?: (string | RegExp)[];
 }
 declare type Pages = PageOption[];
 interface UserOptions {
