diff --git a/node_modules/vite-plugin-html/dist/index.cjs b/node_modules/vite-plugin-html/dist/index.cjs
index 704d055..a4d3403 100644
--- a/node_modules/vite-plugin-html/dist/index.cjs
+++ b/node_modules/vite-plugin-html/dist/index.cjs
@@ -98,7 +98,10 @@ function createPlugin(userOptions = {}) {
         return {
           build: {
             rollupOptions: {
-              input
+              input: {
+                ...input,
+                ...conf.build.rollupOptions.input,
+              },
             }
           }
         };
@@ -118,7 +121,8 @@ function createPlugin(userOptions = {}) {
         _pages = pages.map((page) => {
           return {
             filename: page.filename || DEFAULT_TEMPLATE,
-            template: page.template || DEFAULT_TEMPLATE
+            template: page.template || DEFAULT_TEMPLATE,
+            depFiles: page.depFiles ?? []
           };
         });
       }
@@ -126,30 +130,66 @@ function createPlugin(userOptions = {}) {
       const baseUrl = viteConfig.base ?? "/";
       const keys = Object.keys(proxy);
       let indexPage = null;
+
+      /** @type {{
+        filename: string;
+        depFiles: (string | RegExp)[];
+      }[]} */
+      let reloadEffects = [];
       for (const page of _pages) {
+        reloadEffects.push({
+          filename: page.filename,
+          depFiles: page.depFiles
+        });
         if (page.filename !== "index.html") {
-          rewrites.push(createRewire(page.template, page, baseUrl, keys));
+          rewrites.push(createRewire(baseUrl + page.filename, page, baseUrl, keys));
         } else {
           indexPage = page;
         }
       }
       if (indexPage) {
-        rewrites.push(createRewire("", indexPage, baseUrl, keys));
+        rewrites.push(createRewire(`${baseUrl}(index(.html)?)?`, indexPage, baseUrl, keys));
       }
       server.middlewares.use(history__default({
         disableDotRule: void 0,
         htmlAcceptHeaders: ["text/html", "application/xhtml+xml"],
         rewrites
       }));
+      server.watcher.on('change', file => {
+        reloadEffects.forEach(({ filename, depFiles }) => {
+          if (depFiles.some(depFile => {
+            if (typeof depFile === 'string') {
+              return file.includes(depFile)
+            } else {
+              return depFile.test(file)
+            }
+          })) {
+            server.ws.send({
+              type: 'full-reload',
+              path: filename === 'index.html' ? '*' : `${baseUrl}${filename}*`
+            })
+          }
+        })
+      });
     },
     transformIndexHtml: {
       enforce: "pre",
       async transform(html, ctx) {
-        const url = ctx.filename;
+        const url = ctx.originalUrl ?? ctx.filename;
+
         const base = viteConfig.base;
-        const excludeBaseUrl = url.replace(base, "/");
-        const htmlName = path__default.relative(process.cwd(), excludeBaseUrl);
-        const page = getPage(userOptions, htmlName, viteConfig);
+        let excludeBaseUrl = url;
+
+        if (process.env.NODE_ENV !== 'development') {
+          excludeBaseUrl = path__default.relative(process.cwd(), excludeBaseUrl);
+          excludeBaseUrl = excludeBaseUrl.replace('.html', '');
+        } else {
+          excludeBaseUrl = excludeBaseUrl.replace(base, "")
+        }
+
+        if (excludeBaseUrl === "")
+          excludeBaseUrl = "index.html";
+        const page = getPage(userOptions, excludeBaseUrl, viteConfig);
         const { injectOptions = {} } = page;
         const _html = await renderHtml(html, {
           injectOptions,
@@ -198,13 +238,16 @@ function createPlugin(userOptions = {}) {
 function createInput({ pages = [], template = DEFAULT_TEMPLATE }, viteConfig) {
   const input = {};
   if (isMpa(viteConfig) || pages?.length) {
-    const templates = pages.map((page) => page.template);
-    templates.forEach((temp) => {
-      let dirName = path__default.dirname(temp);
-      const file = path__default.basename(temp);
-      dirName = dirName.replace(/\s+/g, "").replace(/\//g, "-");
-      const key = dirName === "." || dirName === "public" || !dirName ? file.replace(/\.html/, "") : dirName;
-      input[key] = path__default.resolve(viteConfig.root, temp);
+    pages.forEach(({ template, filename }) => {
+      const dirName = path__default
+        .dirname(template)
+        .replace(/\s+/g, "")
+        .replace(/\//g, "-");
+      const key = filename
+        .replace(/\.html/, "")
+        .replace(/\s+/g, "")
+        .replace(/\//g, "-");
+      input[key] = path__default.resolve(viteConfig.root, template);
     });
     return input;
   } else {
@@ -227,7 +270,7 @@ async function renderHtml(html, config) {
     ...viteConfig?.env ?? {},
     ...viteConfig?.define ?? {},
     ...env || {},
-    ...data
+    ...(typeof data === 'function' ? await data() : data)
   };
   let result = await ejs.render(html, ejsData, ejsOptions);
   if (entry) {
@@ -279,13 +322,18 @@ function getPageConfig(htmlName, pages, defaultPage) {
     template: `./${defaultPage}`
   };
   const page = pages.filter((page2) => {
-    return path__default.resolve("/" + page2.template) === path__default.resolve("/" + htmlName);
+    if (process.env.NODE_ENV === 'production') {
+      return page2.filename === htmlName || page2.template === htmlName ||
+             page2.filename === `${htmlName}.html` || page2.template === `${htmlName}.html`;
+    } else {
+      return page2.filename === htmlName;
+    }
   })?.[0];
   return page ?? defaultPageOption ?? void 0;
 }
 function createRewire(reg, page, baseUrl, proxyUrlKeys) {
   return {
-    from: new RegExp(`^/${reg}*`),
+    from: new RegExp(`${reg}.*`),
     to({ parsedUrl }) {
       const pathname = parsedUrl.pathname;
       const excludeBaseUrl = pathname.replace(baseUrl, "/");
diff --git a/node_modules/vite-plugin-html/dist/index.d.ts b/node_modules/vite-plugin-html/dist/index.d.ts
index 0925ff6..b6c8e82 100644
--- a/node_modules/vite-plugin-html/dist/index.d.ts
+++ b/node_modules/vite-plugin-html/dist/index.d.ts
@@ -6,7 +6,7 @@ interface InjectOptions {
     /**
      *  @description Data injected into the html template
      */
-    data?: Record<string, any>;
+    data?: Record<string, any> | (() => Record<string, any>);
     tags?: HtmlTagDescriptor[];
     /**
      * @description esj options configuration
@@ -18,6 +18,7 @@ interface PageOption {
     template: string;
     entry?: string;
     injectOptions?: InjectOptions;
+    depFiles?: (string | RegExp)[];
 }
 declare type Pages = PageOption[];
 interface UserOptions {
